### PR TITLE
Feature/update bazel config

### PR DESCRIPTION
**Pull request by frank30941**
I updated the bazel configuration
- Use bazel version 6.5.0
note:
When there are changes to the go file, it is recommended to run the following command
``` shell
bazelisk run gazelle
```
If add some deps, run
``` shell
bazelisk run gazelle-update-repos
```

### CODE REVIEW

**Review by jeevatkm — CHANGES_REQUESTED** [review]
@frank30941 Thanks for creating a PR.

Please delete these files from the PR.
```
.bazelignore
.bazelversion
WORKSPACE
deps.bzl
```

File: BUILD.bazel [hunk 1]
```diff
@@ -1,10 +1,21 @@
...
+    name = "gazelle-update-repos",
+    args = [
+        "-from_file=go.mod",
+        "-to_macro=deps.bzl%go_dependencies",
+        "-prune",
+        "-build_file_proto_mode=disable_global",
+    ],
+    command = "update-repos",
+)
+
```
Comment (frank30941) [inline BUILD.bazel:18]: @jeevatkm I will delete this code, because it will generate deps.bzl

**Review by jeevatkm — CHANGES_REQUESTED** [review]
@frank30941 Thanks for updating the PR. 

Just one more change is required. Please revert the changes on the following files -
* .gitignore
* go.sum

File: BUILD.bazel (hunk 1 above)
Comment (jeevatkm) [reply BUILD.bazel]: Thanks @frank30941

**Review by jeevatkm — APPROVED** [review]
@frank30941 Thanks for updating the PR.

**Pull request merged**